src/joint_types.jl

```diff
@@ -131,6 +131,80 @@ function velocity_to_configuration_derivative!(q̇::AbstractVector, jt::Quaterni
     nothing
 end
 
+
+"""
+$(SIGNATURES)
+
+Copy the matrix ``source`` to a block of the same size in ``destination``
+spanning the given given range of rows and columns. In Julia v0.7 and above we
+should be able to eliminate this function in favor of:
+
+```julia
+destination[dest_rows, dest_cols] .= source
+```
+
+since the resulting view will be elided.
+"""
+function _copyto_offset!(destination::AbstractMatrix, source::AbstractMatrix, dest_rows::UnitRange{Int} = 1:size(source, 1), dest_cols::UnitRange{Int} = 1:size(source, 2))
+    @boundscheck length(dest_rows) == size(source, 1)
+    @boundscheck length(dest_cols) == size(source, 2)
+    for j in 1:size(source, 2)
+        for i in 1:size(source, 1)
+            destination[dest_rows[i], dest_cols[j]] = source[i, j]
+        end
+    end
+end
+
+
+"""
+$(SIGNATURES)
+
+Compute the jacobian ``Q_v`` which maps joint velocity to configuration
+derivative for the joint type ``jt``:
+
+```math
+\\dot{q} = Q_v v
+```
+
+This updates the entries of Q_v in place. Note that only the structurally
+non-zero entries in V_q are modified, so you may need to initialize V_q
+to zero before first calling this method.
+"""
+function velocity_to_configuration_derivative_jacobian!(Q_v::AbstractMatrix, jt::QuaternionFloating, q::AbstractVector)
+    @boundscheck size(Q_v) == (7, 6)
+    quat = rotation(jt, q)
+    # Q_v[1:4, 1:3] .= velocity_jacobian(quaternion_derivative, quat)
+    # Q_v[5:7, 4:6] .= quat
+    _copyto_offset!(Q_v, velocity_jacobian(quaternion_derivative, quat), 1:4, 1:3)
+    _copyto_offset!(Q_v, quat, 5:7, 4:6)
+    nothing
+end
+
+"""
+$(SIGNATURES)
+
+Compute the jacobian ``V_q`` which maps joint configuration derivative to
```

Comment:
For reference: Duindam used `v = V_Q(\dot{Q})`, i.e. `V_Q` is the linear mapping from configuration derivative to velocity in configuration `Q` (which may not even be vector itself, in Duindam's thesis). So `V_Q` actually isn't a matrix itself. Then he used `V_Q^{-1}` for the inverse mapping.

Let's just introduce new notation for these matrices. How about:

`J_{\dot{q} \rightarrow v}` and `J_{v \rightarrow \dot{q}}`

and then just use `J` as the argument name in code?

Reply:
ok, done